### PR TITLE
[KOGITO-2790] Setup build examples images

### PR DESCRIPTION
**Pull request by radtriste**
https://issues.redhat.com/browse/KOGITO-2790

Many thanks for submiting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.MD#sending-a-pull-request)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a unit test that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -69,11 +71,28 @@ pipeline {
...
+                        // Do not build native images for the PR checks
+                        sh "make build-examples-images tags='~@native' concurrent=1 ${getBDDParameters('never', false)}"
+                    }
+                }
+            }
+            post {
+                always {
+                    archiveArtifacts artifacts: 'test/logs/**/*.log', allowEmptyArchive: true
+                    junit testResults: 'test/logs/**/junit.xml', allowEmptyResults: true
+                    sh "cd test && go run scripts/prune_namespaces.go"
```
Comment (sutaakar) [inline Jenkinsfile:87]: Namespace pruning is not needed for image generation.

File: Jenkinsfile [hunk 2]
```diff
@@ -69,11 +71,28 @@ pipeline {
                 }
             }
         }
+        stage("Build examples' images for testing"){
+            steps {
+                script {
+                    dir ("${WORKING_DIR}") {
+                        // Do not build native images for the PR checks
+                        sh "make build-examples-images tags='~@native' concurrent=1 ${getBDDParameters('never', false)}"
```
Comment (sutaakar) [inline Jenkinsfile:79]: It also needs `cr_deployment_only=true`, though it can be added directly in makefile.....
Will do so.

Comment (sutaakar) [reply Jenkinsfile:79]: Well, actually not needed here as CLI is build before.

Comment (radtriste) [reply Jenkinsfile:79]: https://github.com/kiegroup/kogito-cloud-operator/pull/458

**Comment by radtriste** [comment]
after some discussion, this PR will add the build of examples with mvn locally before building images

**Comment by radtriste** [comment]
jenkins, test this

**Comment by radtriste** [comment]
jenkins, test this

**Comment by radtriste** [comment]
jenkins, test this

**Comment by radtriste** [comment]
jenkins, test this

**Comment by radtriste** [comment]
jenkins, test this

File: Jenkinsfile [hunk 3]
```diff
@@ -10,6 +10,12 @@ pipeline {
         buildDiscarder logRotator(artifactDaysToKeepStr: '', artifactNumToKeepStr: '', daysToKeepStr: '', numToKeepStr: '10')
         timeout(time: 90, unit: 'MINUTES')
     }
+    environment {
+        OPENSHIFT_INTERNAL_REGISTRY = "image-registry.openshift-image-registry.svc:5000"
```
Comment (Kaitou786) [inline Jenkinsfile:14]: Should we be exposing this address?

Comment (radtriste) [reply Jenkinsfile:14]: this is the default internal openshift address so I think we are safe ;)

**Pull request merged**